pkg/image/unprivileged/registry_test.go

```diff
@@ -0,0 +1,113 @@
+package unprivileged
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/docker/distribution/configuration"
+	"github.com/docker/distribution/registry"
+	_ "github.com/docker/distribution/registry/storage/driver/filesystem"
+	_ "github.com/docker/distribution/registry/storage/driver/inmemory"
+	"github.com/phayes/freeport"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/mod/sumdb/dirhash"
+)
+
+func setupRegistry(t *testing.T, ctx context.Context, rootDir string) string {
+	dockerPort, err := freeport.GetFreePort()
+	require.NoError(t, err)
+
+	config := &configuration.Configuration{}
+	config.HTTP.Addr = fmt.Sprintf(":%d", dockerPort)
+	if rootDir != "" {
+		config.Storage = map[string]configuration.Parameters{"filesystem": map[string]interface{}{
+			"rootdirectory": rootDir,
+		}}
+	} else {
+		config.Storage = map[string]configuration.Parameters{"inmemory": map[string]interface{}{}}
+	}
+	config.HTTP.DrainTimeout = time.Duration(2) * time.Second
```

Comment:
Nit: I feel like I usually see the integer literal directly multiplied by the time unit base, but maybe I'm mistaken and that isn't conventional style... the only example in the godoc is written this way. :man_shrugging:

Reply:
I also thought this was strange -- I used oras as an example for this.